packages/typespec-vscode/src/vscode-cmd/import-from-openapi3.ts

```diff
@@ -0,0 +1,333 @@
+import { NodePackage } from "@typespec/compiler";
+import { major, minor } from "semver";
+import vscode from "vscode";
+import logger from "../log/logger.js";
+import { normalizePath } from "../path-utils.js";
+import { Result, ResultCode } from "../types.js";
+import {
+  checkAndConfirmEmptyFolder,
+  selectFile,
+  selectFolder,
+  tryExecuteWithUi,
+} from "../ui-utils.js";
+import {
+  ExecOutput,
+  isDirectory,
+  isExecOutputCmdNotFound,
+  loadDependencyPackageJson,
+  searchAndLoadPackageJson,
+  spawnExecutionAndLogToOutput,
+} from "../utils.js";
+
+const TITLE = "Import TypeSpec From OpenAPI3";
+const TSP_OPENAPI3_COMMAND = "tsp-openapi3";
+const TSP_COMPILER_PACKAGE = "@typespec/compiler";
+const TSP_OPENAPI3_PACKAGE = "@typespec/openapi3";
+const TSP_OPENAPI3_PACKAGE_DETAILS =
+  "TypeSpec library for emitting OpenAPI 3.0 from the TypeSpec REST protocol binding and converting OpenAPI3 to TypeSpec";
+const TSP_OPENAPI3_PACKAGE_LINK = "https://typespec.io/docs/emitters/openapi3/cli/";
+
+export async function importFromOpenApi3(uri: vscode.Uri | undefined) {
+  await vscode.window.withProgress(
+    {
+      location: vscode.ProgressLocation.Window,
+      cancellable: false,
+      title: "Importing from OpenAPI...",
+    },
+    async () => {
+      const targetFolder =
+        uri && uri.fsPath && (await isDirectory(uri.fsPath))
+          ? normalizePath(uri.fsPath)
+          : await selectFolder("Select target folder to import OpenAPI", "Select Target Folder");
+      if (!targetFolder) {
+        logger.info("Importing from OpenApi canceled because no target folder selected");
+        return;
+      }
+      const checkEmpty = await checkAndConfirmEmptyFolder(
+        targetFolder,
+        "Selected folder is not empty. Do you want to continue?",
```

Comment:
Should we be explicit about what could happen here? For example - that some files may be overwritten if present?

Reply:
Update to "The selected folder is not empty. Some files may be overwritten if they already exist. Are you sure you want to import here?"